lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -140,6 +144,97 @@ class AddNewPrimUndoableCommand : public Ufe::UndoableCommand
     PXR_NS::TfToken _primToken;
 };
 
+const char* selectUSDFileScript = R"(
+global proc string SelectUSDFileForAddReference()
+{
+    string $result[] = `fileDialog2 
+        -fileMode 1
+        -caption "Add Reference to USD Prim"
+        -fileFilter "USD Files (*.usd *.usda *.usdc);;*.usd;;*.usda;;*.usdc"`;
+
+    if (0 == size($result))
+        return "";
+    else
+        return $result[0];
+}
+SelectUSDFileForAddReference();
+)";
+
+const char* clearAllReferencesConfirmScript = R"(
+global proc string ClearAllUSDReferencesConfirm()
+{
+    return `confirmDialog -title "Remove All References" 
+        -message "Removing all references from USD prim.  Are you sure?"
+        -button "Yes" -button "No" -defaultButton "Yes"
+        -cancelButton "No" -dismissString "No"`;
+
+}
+ClearAllUSDReferencesConfirm();
+)";
+
+class AddReferenceUndoableCommand : public Ufe::UndoableCommand
+{
+public:
+    static const std::string commandName;
+
+    AddReferenceUndoableCommand(
+        const UsdPrim& prim,
+        const std::string& filePath
+    ) : fPrim(prim),
+        fSdfRef(),
+        fFilePath(filePath)
+    {}
+
+    void undo() override { 
+        if (fPrim.IsValid()) {
+            UsdReferences primRefs = fPrim.GetReferences();
+            primRefs.RemoveReference(fSdfRef);
+        }
+    }
+
+    void redo() override { 
+        if (fPrim.IsValid()) {
+            fSdfRef = SdfReference(fFilePath);
+            UsdReferences primRefs = fPrim.GetReferences();
+            primRefs.AddReference(fSdfRef);
+        }
+    }
+
+private:
+    UsdPrim fPrim;
+    SdfReference fSdfRef;
+    const std::string fFilePath;
+};
+const std::string AddReferenceUndoableCommand::commandName("Add Reference...");
+
+class ClearAllReferencesUndoableCommand : public Ufe::UndoableCommand
+{
+public:
+    static const std::string commandName;
+    static const MString cancelRemoval;
+
+    ClearAllReferencesUndoableCommand(
+        const UsdPrim& prim
+    ) : fPrim(prim)
+    {}
+
+    void undo() override { 
+
```

Comment:
Not currently providing an Undo for clearing all references.

Reply:
Is it because it's technically difficult, or because you're waiting for design to think about it?  Perhaps we should add a separate JIRA entry for this one, so that it doesn't fall through the cracks.

Reply:
Bit of both actually.  The request here was for something pretty basic, really just enough to get started with testing how the final design should look/work.  If I'd seen a really obvious way of dealing with it then I would have added it, but it doesn't look like there is a straight forward get'er for the complete list of references to be stored (and then put back as part of Undo).